packages/expo-file-system/ios/EXFileSystem/EXFileSystemAssetLibraryHandler.m

```diff
@@ -66,15 +66,27 @@ + (void)copyFrom:(NSURL *)from
 
   if (fetchResult.count > 0) {
     PHAsset *asset = fetchResult[0];
-    [[PHImageManager defaultManager] requestImageDataForAsset:asset options:nil resultHandler:^(NSData * _Nullable imageData, NSString * _Nullable dataUTI, UIImageOrientation orientation, NSDictionary * _Nullable info) {
-      if ([imageData writeToFile:toPath atomically:YES]) {
-        resolve(nil);
-      } else {
-        reject(@"E_FILE_NOT_COPIED",
-               [NSString stringWithFormat:@"File '%@' could not be copied to '%@'.", from, to],
-               error);
-      }
-    }];
+    if (asset.mediaType == PHAssetMediaTypeVideo) {
+      [[PHImageManager defaultManager] requestAVAssetForVideo:asset options:nil resultHandler:^(AVAsset *asset, AVAudioMix *audioMix, NSDictionary *info) {
+        if (![asset isKindOfClass:[AVURLAsset class]]) {
+          reject(@"ERR_INCORRECT_ASSET_TYPE",
+                 [NSString stringWithFormat:@"File '%@' has incorrect asset type.", from],
```

Comment:
would this be an incorrect asset type from the user? If so, can we include what the expected asset type is in the error message?

Reply:
It should be only triggered when the system doesn't have the URL for the asset - for example, if it's store in the cloud. But if that is the case, I don't think you can obtain a valid URL - so the copy function shouldn't be triggered without downloading the asset itself.

Reply:
I see, so it's unlikely a user would run into this error? I bring it up because if I received the error message `File <my file> has incorrect asset type.` as a user, my first question would be "What should I do differently?"

Reply:
Yes, it'll be very unlikely.